src/main/java/org/sagebionetworks/web/client/widget/entity/file/FileDownloadButton.java

```diff
@@ -198,8 +216,32 @@ public void onFailure(Throwable caught) {
 	}
 	
 	@Override
-	public void onLicensedDownloadClick() {
-		//will be re-using for any js handling of download (s3 direct download, for example!)
+	public void onUnauthenticatedS3DirectDownloadClicked() {
+		//ask for credentials, use bucket/endpoint info from storage location
+		ExternalObjectStoreFileHandle objectStoreFileHandle = (ExternalObjectStoreFileHandle) dataFileHandle;
+		view.showLoginS3DirectDownloadDialog(objectStoreFileHandle.getEndpointUrl());
+	}
+	
+	@Override
+	public void onLoginS3DirectDownloadClicked(String accessKeyId, String secretAccessKey) {
+		final ExternalObjectStoreFileHandle objectStoreFileHandle = (ExternalObjectStoreFileHandle) dataFileHandle;
+		CallbackP<JavaScriptObject> s3Callback = new CallbackP<JavaScriptObject>() {
+			@Override
+			public void invoke(JavaScriptObject s3JsObject) {
+				s3 = s3JsObject;
+				// NOTE: most browsers block the popup because the button click event is not directly associated to the login popup.
+				// Show the direct download button after authorization succeeds.
+				view.showS3DirectDownloadDialog();
+			}
+		};
+		awsSdk.getS3(accessKeyId, secretAccessKey, objectStoreFileHandle.getBucket(), objectStoreFileHandle.getEndpointUrl(), s3Callback);	
+	}
+	
+	@Override
+	public void onAuthenticatedS3DirectDownloadClicked() {
+		ExternalObjectStoreFileHandle objectStoreFileHandle = (ExternalObjectStoreFileHandle) dataFileHandle;
+		String presignedUrl = awsSdk.getPresignedURL(objectStoreFileHandle.getFileKey(), objectStoreFileHandle.getBucket(), s3);
+		popupUtilsView.openInNewWindow(presignedUrl);
```

Comment:
@zimingd , is there a way to tell Synapse (for auditing purposes) that the user has downloaded one of these new external object store files?